www/modules/custom/borg_project_metrics/borg_project_metrics.module

```diff
@@ -84,4 +90,284 @@ function borg_project_metrics_cron() {
       }
     }
   }
+
+  // Get and store backdrop_project_metrics top level data weekly.
+  if (date('G') == 24) {
+    if (date('l') == 'Monday') {
+      try {
+        $last_date = _get_display_metric_from_db('commits')['date'];
+        _parse_metrics('commits', $last_date);
+      }
+      catch(Exception $e) {
+        $message = $e->getMessage();
+        watchdog('borg_project_metrics', $message, array(), WATCHDOG_ERROR);
+      }
+    }
+
+    if (date('l') == 'Tuesday') {
+      try {
+        $last_date = _get_display_metric_from_db('comments')['date'];
+        _parse_metrics('comments', $last_date);
+      }
+      catch(Exception $e) {
+        $message = $e->getMessage();
+        watchdog('borg_project_metrics', $message, array(), WATCHDOG_ERROR);
+      }
+    }
+  }
+}
+
+/**
+ * Get the number of commits to Backdrop since 15 January 2015.
+ *
+ * @param string $type
+ * String to indicate what metric to get: {users, commits, comments}.
+ *
+ * @param string $date_time
+ * The date from DB borg_project_metrics table to set as since for the get.
+ *
+ * @return array $metrics
+ * Returns an array of json objects each element is a paginated response from the
+ * github api.
+ *
+ * @see _get_contrib_comments()
+ */
+function _get_metrics(string $type, string $date_time = '2017-06-01T00:00:00Z') {
+  global $settings;
+  $token = $settings['githubapi_token'];
+  $authorization = "Authorization: token $token";
+  $metrics = [];
+  $i = 1;
+  $date_time = str_replace(' ', 'T', $date_time) . 'Z';
+  do {
+    if ($type == 'users') {
+      $url = "https://api.github.com/repos/backdrop/backdrop/commits?since=$date_time&page=$i";
+    }
+    elseif ($type == 'commits') {
+      $url = "https://api.github.com/repos/backdrop/backdrop/commits?since=$date_time&page=$i";
+    }
+    elseif ($type = 'comments') {
+      $url = "https://api.github.com/repos/backdrop/backdrop-issues/issues/comments?since=$date_time&page=$i";
+    }
+
+    // Initiate a curl session.
+    $ch = curl_init();
+    curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/json', $authorization));
+    curl_setopt ($ch, CURLOPT_USERAGENT, "Mozilla/4.0 (compatible; MSIE 5.01; Windows NT 5.0)");
+    curl_setopt($ch, CURLOPT_URL, $url);
+    curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+    curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 1);
+    $metrics[$i] = curl_exec($ch);
+    $res_code = curl_getinfo($ch, CURLINFO_RESPONSE_CODE);
+    curl_close($ch);
+    $json = json_decode($metrics[$i]);
+    $i++;
+  } while ($res_code == 200 && !empty($json));
+
+  // If the $type is comments then we also want to get the contrib comments.
```

Comment:
My opinion is we should get contrib commits too, but I think we should get this in as is and work on that as a separate issue.